code/game/objects/items/RCD.dm

```diff
@@ -61,47 +64,69 @@ RLD
 	silo_mats = null
 	return ..()
 
+/obj/item/construction/melee_attack_chain(mob/user, atom/target, params)
+	if(istype(target, /obj/item/rcd_upgrade))
+		install_upgrade(target, user)
+		return
+	if(insert_matter(target, user))
+		return
+	return ..()
+
 /obj/item/construction/attackby(obj/item/W, mob/user, params)
-	if(iscyborg(user))
+	if(istype(W, /obj/item/rcd_upgrade))
+		install_upgrade(W, user)
+		return
```

Comment:
Do you want to continue the attack if you installed the upgrade or inserted matter? If not, return truthy values.